src/mono/wasm/runtime/es6/dotnet.es6.lib.js

```diff
@@ -121,7 +121,12 @@ const linked_functions = [
 // -- this javascript file is evaluated by emcc during compilation! --
 // we generate simple proxy for each exported function so that emcc will include them in the final output
 for (let linked_function of linked_functions) {
+    #if USE_PTHREADS
+    const fn_template = `return __dotnet_runtime.__linker_exports.${linked_function}.apply(__dotnet_runtime, arguments)`;
+    DotnetSupportLib[linked_function] = new Function(fn_template);
+    #else
     DotnetSupportLib[linked_function] = new Function("");
```

Comment:
Can we make this `throw new Error("unreachable")` instead? It was clearly a mistake for me to approve the existing no-op function, we don't seem to expect these to ever be invoked so we should make sure they're not. Otherwise we have silent failure like before.

Reply:
All this should do is to fool emscripten't linker. It will endup in here
![image](https://user-images.githubusercontent.com/271576/215156924-d2df370d-3f56-4368-999e-9282cba6010e.png)
And so it's bit pointless to spill download bytes of the functions as well as their names.
But yeah, I will add throw for now.